src/utils/vector.cuh

```diff
@@ -370,6 +370,12 @@ VECATTR  float4 floorf(const float4 &a){
 
 VECATTR float dot(float4 a, float4 b){return a.x*b.x + a.y*b.y + a.z*b.z + a.w*b.w;}
 
+VECATTR float4 normalize(float4 v)
+{
+  float invLen = 1.0/sqrt(dot(v, v));
```

Comment:
You can use float invLen = rsqrtf(dot(v, v));
Also, as it is now this code uses double precision (1.0 and sqrt)